src/taipy/core/config/migration_config.py

```diff
@@ -0,0 +1,192 @@
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import collections.abc
+from copy import deepcopy
+from typing import Any, Callable, Dict, Optional, Union
+
+from taipy.config._config import _Config
+from taipy.config.common._template_handler import _TemplateHandler as _tpl
+from taipy.config.config import Config
+from taipy.config.unique_section import UniqueSection
+
+from .._version._version_manager import _VersionManager
+from ..common.alias import DataNodeId, PipelineId, ScenarioId, TaskId
+from .data_node_config import DataNodeConfig
+from .pipeline_config import PipelineConfig
+from .scenario_config import ScenarioConfig
+from .task_config import TaskConfig
+
+
+class MigrationConfig(UniqueSection):
+    """
+    Configuration fields needed to register migration functions from an old version to newer one.
+
+    Attributes:
+        migration_fcts (Dict[str, Dict[str, Callable]]): A dictionary that maps the version that entities are
+            migrated from to the migration functions.
+        **properties (dict[str, Any]): A dictionary of additional properties.
+    """
+
+    name = "VERSION_MIGRATION"
+
+    _MIGRATION_FCTS_KEY = "migration_fcts"
+
+    def __init__(
+        self,
+        migration_fcts: Dict[str, Dict[str, Callable]],
+        **properties,
+    ):
+        self.migration_fcts = migration_fcts
+
+        super().__init__(**properties)
+
+    def __copy__(self):
+        return MigrationConfig(
+            deepcopy(self.migration_fcts),
+            **deepcopy(self._properties),
+        )
+
+    def __getattr__(self, item: str) -> Optional[Any]:
+        return _tpl._replace_templates(self._properties.get(item))  # type: ignore
+
+    @classmethod
+    def default_config(cls):
+        return MigrationConfig({})
+
+    def _to_dict(self):
+        return {
+            self._MIGRATION_FCTS_KEY: self.migration_fcts,
+            **self._properties,
+        }
+
+    @classmethod
+    def _from_dict(cls, as_dict: Dict[str, Any], id: str, config: Optional[_Config]):
+        return MigrationConfig(**as_dict)
+
+    def _update(self, as_dict, default_section=None):
+        def deep_update(d, u):
+            for k, v in u.items():
+                if isinstance(v, collections.abc.Mapping):
+                    d[k] = deep_update(d.get(k, {}), v)
+                else:
+                    d[k] = v
+            return d
+
+        migration_fcts = as_dict.pop(self._MIGRATION_FCTS_KEY)
+        for source_version in list(migration_fcts.keys()):
+            if (proper_version := _VersionManager._replace_version_number(source_version)) != source_version:
```

Comment:
I missed this before, but I have the same misunderstanding here. Why do we use `_VersionManager._replace_version_number(source_version)` and not `source_version` directly

Reply:
We don't need to replace the version.